### PR TITLE
Update documentation to include info on the Exporter

### PR DESCRIPTION
**Pull request by NMe84**
I needed to use the export functionality that @MaximePinot gracefully [added](https://github.com/omines/datatables-bundle/pull/83) a while ago and noticed that there was no documentation for it. I got to work with the information given in the PR and what I could gather from the code itself and wrote something that I think should be clear enough to get people going.

Maxime, would you be willing to review this?

### CODE REVIEW

File: docs/source/index.html.md [hunk 1]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
 which is fulfilled with the `DataTables` instance once initialization is completed. This allows you
 all the flexibility you could need to [invoke API functions](https://datatables.net/reference/api/).
 
+# Exporting Data
+
+DataTables natively supports exporting the current view as an Excel sheet, PDF, etc. The problem with this
+is that this is all generated on the client's end meaning you'll be unable to export all data at once unless it
```
Comment (MaximePinot) [inline docs/source/index.html.md:578]:
```suggestion
is that this is all generated on the client's end, meaning you'll be unable to export all data at once unless it
```

File: docs/source/index.html.md [hunk 2]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+happens to fit on a single screen. This is why this bundle supports exporting data from the server's end instead.
+
+## Prerequisites
+
+To make use of the exporting functionality in this bundle there are some extra dependencies you may want to install.
```
Comment (MaximePinot) [inline docs/source/index.html.md:583]:
```suggestion
To make use of the exporting functionality in this bundle, there are some extra dependencies you may want to install.
```

File: docs/source/index.html.md [hunk 3]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+These are optional as you may not need the export feature at all, or you might not need all options available.
+
+The easiest way to make the export feature work on the client side is by using the [Buttons extension](https://datatables.net/extensions/buttons/)
+offered by `DataTables`.
```
Comment (MaximePinot) [inline docs/source/index.html.md:587]: The `Buttons` extension is required, so I would emphasize that it is a required dependency to make the server-side export feature work.

  ```suggestion
The server-side export feature requires the [Buttons extension](https://datatables.net/extensions/buttons/) offered by `DataTables`. Make sure to install it before using this feature.
```

File: docs/source/index.html.md [hunk 4]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+        dom:'<"html5buttons"B>lTfgitp',
+        buttons: [
+            {
+                text: 'Export to Excel',
+                action: $.fn.initDataTables.exportBtnAction('excel', {{ datatable_settings(datatable) }})
+            }
+        ]
+    });
+});
+```
```
Comment (MaximePinot) [inline docs/source/index.html.md:629]: IMO, here are some things that are important to be mentioned here:

- One must use the `$.fn.initDataTables.exportBtnAction` helper function to init the button
- The first parameter is the name of the exporter (see: `DataTableExporterInterface::getName`)

File: docs/source/index.html.md [hunk 5]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+Currently included in the bundle there are a `DataTableExporter` for Excel spreadsheets and one for CSV. If you'd like
+to use the Excel exporter you will need to add [phpoffice/phpspreadsheet](https://packagist.org/packages/phpoffice/phpspreadsheet)
+to your dependencies.
+
+## Examples
+
+### Server Side Setup
+
+On the server side of things all you will need to add is an event listener that will handle passing the resulting file
+back to the user:
```
Comment (MaximePinot) [inline docs/source/index.html.md:598]: :warning:  The event listener is optional.

Only listen to the event if you need to modify the generated file (e.g. changing the auto-generated name.)

File: docs/source/index.html.md [hunk 6]
```diff
@@ -572,6 +572,69 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+    });
+});
+```
+
+## Custom Exporters
+
+Excel and CSV are nice but there are other formats you may want to export to. You can support this by implementing
+the [Omines\DataTablesBundle\Exporter\DataTableExporterInterface](https://github.com/omines/datatables-bundle/blob/master/src/Exporter/DataTableExporterInterface.php)
+in a class of your own. You will need to register this class as service in Symfony as well and the service
+needs to be tagged with `datatables.exporter`.
```
Comment (MaximePinot) [inline docs/source/index.html.md:636]:
```suggestion
needs to be tagged with `datatables.exporter` (this should be done automatically if you have [autoconfigure](https://symfony.com/doc/current/service_container.html#the-autoconfigure-option) turned on).
```

**Review by MaximePinot — COMMENTED** [review]
Hi @NMe84,

This is great! Thank you for taking the time to write some documentation about this feature.

I added some comments with the things that I believe should be worth mentioning.

**Comment by NMe84** [comment]
Thanks! I applied all the suggested changes you made and I'll look into incorporating the other two suggestions into text tomorrow.

**Comment by NMe84** [comment]
I think [this commit](https://github.com/omines/datatables-bundle/pull/239/commits/c8b4426dfe1984f6d13e19033a979aab167ff34c) should cover the two currently open conversations, unless I messed up my wording.

File: docs/source/index.html.md [hunk 7]
```diff
@@ -572,6 +572,71 @@ The function returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/
...
+## Examples
+
+### Server Side Setup
+
+On the server side of things you don't need to do anything besides the setup mentioned above. You can however customize some
+things like the filename of the file that is being generated. To do this, all you will need to add is an event listener
+that will handle passing the resulting file back to the user.
+
+```php
+$table = $this->createDataTable()
```
Comment (MaximePinot) [inline docs/source/index.html.md:599]: The `DataTablesTrait` is deprecated.

```suggestion
$table = $dataTableFactory->create()
```

**Comment by MaximePinot** [comment]
It looks good to me! :+1: Nice work!

**Comment by NMe84** [comment]
Cool, thanks! Let's see what the big boss has to say. 😄 @curry684 ?

**Comment by curry684** [comment]
Big boss says awesoooooome   😉 

Now I just have to fix the doc generator haha.

**Pull request merged**